src/metabase/driver/common/parameters/dates.clj

```diff
@@ -539,14 +541,16 @@
   [raw-date-str field-type]
   (let [;; `raw-date-str` is sanitized in case it contains millis and timezone which are incompatible
         ;; with [[date-string->range]]. `substitute-field-filter-test` expects that to happen.
-        range-raw (try (date-string->range raw-date-str)
-                       (catch Throwable _
-                         (fallback-raw-range raw-date-str)))
+        [range-raw unit] (try (let [r (date-string->raw-range raw-date-str)]
+                                [(format-date-range r) (:unit r)])
+                              (catch Throwable _
+                                [(fallback-raw-range raw-date-str) :day]))
         date-str-conversion (if (isa? field-type :type/DateTimeWithTZ)
                               date-str->qp-aware-offset-dt
                               date-str->local-dt)]
     (-> (update-vals range-raw date-str-conversion)
-        (m/update-existing :end exclusive-datetime-range-end (date-str->unit-fn (:end range-raw)))
+        (m/update-existing :end exclusive-datetime-range-end (or ({:second t/seconds, :hour t/hours} unit)
```

Comment:
Only seconds and hours have the reported bug, and from the problem with seconds cannot be triggered from the FE, but I handle it for completeness.